lib/src/message_list_view.dart

```diff
@@ -518,15 +531,20 @@ class _MessageListViewState extends State<MessageListView> {
           .map((threads) => threads[widget.parentMessage.id]);
     }
 
-    _streamListener = stream
-        .map((messages) =>
-            messages
-                ?.where((m) =>
-                    !(m.status == MessageSendingStatus.FAILED && m.isDeleted) &&
-                    m.shadowed != true)
-                ?.toList() ??
-            [])
-        .listen((newMessages) {
+    _streamListener = stream.map((messages) {
+      final filteredMessages = messages
+              ?.where((m) =>
+                  !(m.status == MessageSendingStatus.FAILED && m.isDeleted) &&
+                  m.shadowed != true)
+              ?.toList() ??
+          [];
+
+      if (widget.messageFilter != null) {
+        return messages.where(widget.messageFilter).toList();
+      }
+
+      return filteredMessages;
```

Comment:
I think we should not apply our own filters if there is `widget.messageFilter` provided.